operations/ftdc.go

```diff
@@ -0,0 +1,273 @@
+				Name:  "prefix",
+				Usage: "prefix for FTDC filenames",
+			},
+			cli.IntFlag{
+				Name:  "maxChunkSize",
+				Usage: "maximum chunk size",
+				Value: 1000,
+			},
+			cli.DurationFlag{
+				Name:  "flush",
+				Usage: "flush interval",
+				Value: 20000000,
+			},
+		},
+		Before: mergeBeforeFuncs(
+			requireFileExists("input", true),
+			requireStringFlag("prefix"),
+		),
+		Action: func(c *cli.Context) error {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			opts := ftdc.CollectJSONOptions{}
+
+			jsonPath := c.String("input")
+			if jsonPath == "" {
+				opts.InputSource = os.Stdin
+			} else {
+				opts.FileName = jsonPath
+			}
+			opts.OutputFilePrefix = c.String("prefix")
+			opts.FlushInterval = c.Duration("flush")
+			opts.ChunkSizeBytes = c.Int("maxChunkSize")
+
+			if err := ftdc.CollectJSONStream(ctx, opts); err != nil {
+				return errors.Wrap(err, "Failed to write FTDC from JSON")
+			}
+			return nil
+		},
+	}
+}
+
+func toBSON() cli.Command {
+	return cli.Command{
+		Name:  "tobson",
+		Usage: "write FTDC data to a BSON file",
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "input",
+				Usage: "write FTDC data from this file",
+			},
+			cli.StringFlag{
+				Name:  "output",
+				Usage: "write FTDC data in BSON format to this file (default: stdout)",
+			},
+			cli.BoolFlag{
+				Name:  "flattened",
+				Usage: "flatten FTDC data",
+			},
+		},
+		Before: mergeBeforeFuncs(
+			requireFileExists("input", false),
+		),
+		Action: func(c *cli.Context) error {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			ftdcPath := c.String("input")
+			bsonPath := c.String("output")
+
+			ftdcFile, err := os.Open(ftdcPath)
+			if err != nil {
+				return errors.Wrapf(err, "problem opening file '%s'", ftdcPath)
+			}
+			defer ftdcFile.Close()
+
+			var bsonFile *os.File
+			if bsonPath == "" {
+				bsonFile = os.Stdout
+			} else {
+				bsonFile, err = os.Create(bsonPath)
+				if err != nil {
+					return errors.Wrapf(err, "problem opening flie '%s'", bsonPath)
+				}
+				defer func() { grip.Warning(bsonFile.Close()) }()
+			}
+
+			var iter ftdc.Iterator
+			if c.Bool("flattened") {
+				iter = ftdc.ReadMetrics(ctx, ftdcFile)
+			} else {
+				iter = ftdc.ReadStructuredMetrics(ctx, ftdcFile)
+			}
+
+			for iter.Next(ctx) {
+				bytes, err := iter.Document().MarshalBSON()
+				if err != nil {
+					return errors.Wrap(err, "problem marshaling BSON")
+				}
+				bsonFile.Write(bytes)
+			}
+
+			return errors.Wrap(err, "problem iterating ftdc file")
+		},
+	}
+}
+
+func fromBSON() cli.Command {
+	return cli.Command{
+		Name:  "frombson",
+		Usage: "write FTDC data from a BSON file",
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "input",
+				Usage: "write BSON data from this file",
+			},
+			cli.StringFlag{
+				Name:  "output",
+				Usage: "write BSON data in FTDC format to this file",
+			},
+			cli.IntFlag{
+				Name:  "maxChunkSize",
```

Comment:
should change to say "maxCount" with docs that explain that it's the number of samples per chunk. The value is correct, I think.